WordPress/gutenberg
# Add navigation overlay block

## What?
Adds a block that wraps the navigation overlay.

## Why?
This will be useful for two reasons:

1. It allows us to style the navigation overlay using theme.json
2. It will be useful in allowing us to customize the navigation overlay as per https://github.com/WordPress/gutenberg/issues/43852

## How?
Creates a new block called Navigation Overlay, which is used as the navigation overlay.

## Testing Instructions
1. Switch to a block theme
3. Add this to the theme.json for your theme:
```
			"core/navigation-overlay": {
				"color": {
					"background": "blue",
					"text": "#00ff00"
				}
			},
```
4. Add a navigation block and set it to collapse (the default setting)
<img width="270" alt="Screenshot 2023-10-24 at 11 51 42" src="https://github.com/WordPress/gutenberg/assets/275961/1d93c1d8-e117-44d2-94a7-ec7848d9ad8e">
5. Open the navigation overlay and confirm that the colors from the theme.json are applied:

<img width="582" alt="Screenshot 2023-10-24 at 11 52 24" src="https://github.com/WordPress/gutenberg/assets/275961/4b82e746-180b-4d4e-8814-9cc7731804fc">
